To: vim_dev@googlegroups.com
Subject: Patch 7.3.361
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.361
Problem:    Accessing memory after it is freed when EXITFREE is defined.
Solution:   Don't access curwin when firstwin is NULL. (Dominique Pelle)
Files:	    src/buffer.c

*** ../vim-7.3.360/src/buffer.c	2011-09-14 17:50:05.000000000 +0200
--- src/buffer.c	2011-11-30 15:38:00.000000000 +0100
***************
*** 567,574 ****
      diff_buf_delete(buf);	    /* Can't use 'diff' for unloaded buffer. */
  #endif
  #ifdef FEAT_SYN_HL
!     if (curwin->w_buffer == buf)
! 	reset_synblock(curwin);	    /* remove any ownsyntax */
  #endif
  
  #ifdef FEAT_FOLDING
--- 567,575 ----
      diff_buf_delete(buf);	    /* Can't use 'diff' for unloaded buffer. */
  #endif
  #ifdef FEAT_SYN_HL
!     /* Remove any ownsyntax, unless exiting. */
!     if (firstwin != NULL && curwin->w_buffer == buf)
! 	reset_synblock(curwin);
  #endif
  
  #ifdef FEAT_FOLDING
*** ../vim-7.3.360/src/version.c	2011-11-30 15:19:25.000000000 +0100
--- src/version.c	2011-11-30 15:40:22.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     361,
  /**/


-- 
From the classified section of a city newspaper:
Dog for sale: eats anything and is fond of children.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
